.github/actions/install-all-deps/action.yml

```diff
@@ -37,24 +65,34 @@ runs:
       with:
         path: |
           venv/*
-        key: gradio-lib-ubuntu-latest-pip-${{ hashFiles('client/python/requirements.txt') }}-${{ hashFiles('requirements.txt') }}-${{ hashFiles('test/requirements.txt') }}
+          client/python/venv
+        restore-keys: |
+          gradio-lib-${{inputs.os}}-latest-pip-
+        key: gradio-lib-${{inputs.os}}-latest-pip-${{ hashFiles('client/python/requirements.txt') }}-${{ hashFiles('requirements.txt') }}-${{ hashFiles('test/requirements.txt') }}-${{ hashFiles('client/python/test/requirements.txt') }}
     - name: Install Gradio and Client Libraries Locally (Linux)
       shell: bash
       run: |
-        . venv/bin/activate
-        python -m pip install -e client/python
-        python -m pip install -e .
+        . ${{ env.VENV_ACTIVATE }}
+        python -m pip install client/python  
```

Comment:
I think we need to keep the editable install here for the functional tests?d

Reply:
thanks!